[webkitpy] Move default CURRENT_VERSION variable definition to DarwinPort.

bls1999 opened the pull request:
#### 71871b592a2fe399d0e1c041de1225a504ff66a6
<pre>
[webkitpy] Move default CURRENT_VERSION variable definition to DarwinPort.
<a href="https://bugs.webkit.org/show_bug.cgi?id=294287">https://bugs.webkit.org/show_bug.cgi?id=294287</a>
<a href="https://rdar.apple.com/153002661">rdar://153002661</a>

Reviewed by Jonathan Bedard.

Now that all Apple port versions are 26, we can synthesize all of the different
CURRENT_VERSION definitions by port into one definition in DarwinPort. If we
have different versioning for whatever reason (maybe differences in minor
version?), we can always override that in the subclasses.

This change also changes associated unit tests to use the current version
rather than manually defining it every time, requiring fewer changes each year.

* Tools/Scripts/webkitpy/common/version_name_map.py:
    (VersionNameMap)
        -&gt; (__init__): Pull CURRENT_VERSION from port objects for iOS, tvOS, watchOS, and visionOS.
                        (Not for macOS, as this requires a manual change to define new version names yearly anyway.)
* Tools/Scripts/webkitpy/port/darwin.py:
    (DarwinPort): Set default for CURRENT_VERSION to `Version(26)`.
* Tools/Scripts/webkitpy/port/ios.py:
    (IOSPort): Remove CURRENT_VERSION definition that overrides DarwinPort.
* Tools/Scripts/webkitpy/port/mac.py:
    (MacPort): Ditto.
* Tools/Scripts/webkitpy/port/visionos.py:
    (VisionOSPort): Ditto.
* Tools/Scripts/webkitpy/port/watch.py:
    (WatchPort): Ditto.
* Tools/Scripts/webkitpy/port/ios_device_unittest.py: Change unit tests to use CURRENT_VERSION from associated ports.
* Tools/Scripts/webkitpy/port/ios_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/ios_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/visionos_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_testcase.py: Ditto.

Canonical link: <a href="https://commits.webkit.org/296346@main">https://commits.webkit.org/296346@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/28c5ad3ddd3cb6182b78fcbcfefcb5cc8d6c9b31

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107455 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27137 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17544 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112669 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57991 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109419 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27821 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35637 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81568 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110384 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22030 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96844 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61950 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/106860 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21470 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14988 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57435 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91399 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15015 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115770 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34522 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25437 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90606 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34897 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93099 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90352 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35258 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13041 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30265 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17483 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34443 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34189 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37544 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35850 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/webkitpy/common/version_name_map.py:
```diff
@@ -48,6 +48,8 @@ def __init__(self, platform=None):
             platform = SystemHost.get_default().platform
         self.mapping = {}
 
+        from webkitpy.port import darwin, ios, watch, visionos
```
Is there reason not to put this at the top of the file? Does it just lead to a circular import if we do?

bls1999 replied:
Yep

gsnedders commented on Tools/Scripts/webkitpy/common/version_name_map.py:
```diff
@@ -69,10 +71,10 @@ def __init__(self, platform=None):
             },
-            'ios': self._automap_to_major_version('iOS', minimum=Version(10), maximum=Version(26)),
-            'tvos': self._automap_to_major_version('tvOS', minimum=Version(10), maximum=Version(26)),
-            'watchos': self._automap_to_major_version('watchOS', minimum=Version(1), maximum=Version(26)),
-            'visionos': self._automap_to_major_version('visionOS', minimum=Version(1), maximum=Version(26)),
+            'ios': self._automap_to_major_version('iOS', minimum=Version(10), maximum=ios.IOSPort.CURRENT_VERSION),
+            'tvos': self._automap_to_major_version('tvOS', minimum=Version(10), maximum=darwin.DarwinPort.CURRENT_VERSION),
+            'watchos': self._automap_to_major_version('watchOS', minimum=Version(1), maximum=watch.WatchPort.CURRENT_VERSION),
+            'visionos': self._automap_to_major_version('visionOS', minimum=Version(1), maximum=visionos.VisionOSPort.CURRENT_VERSION),
```
To me, this just drives home how strange this class actually is, largely having logic I'd expect to be owned by the individual ports.

That said, I wonder if `darwin.DarwinPort` is the right thing to use for tvOS? I feel like this is strange insofar as we don't actually have a tvOS port (https://bugs.webkit.org/show_bug.cgi?id=273261); I wonder if it makes more sense to just use the iOS port version?

bls1999 replied:
I can change it to the iOS port, but I think long-term we probably just want to action that bug and add a tvOS port. We lose `CURRENT_VERSION` overridability at the port level without a dedicated tvOS port.

bls1999 replied:
On second look, I'm actually not sure if we should change it. A tvOS port would be a subclass of `DevicePort`, which subclasses `DarwinPort`. Changing to `IOSPort.CURRENT_VERSION` could indicate that we one day intend a tvOS port to extend `IOSPort` instead.

gsnedders reviewed (commented):
This feels like a net improvement; only real question I have here is about the VersionNameMap stuff?

bls1999 committed:
[webkitpy] Move default CURRENT_VERSION variable definition to DarwinPort.
https://bugs.webkit.org/show_bug.cgi?id=294287
rdar://153002661

Reviewed by Jonathan Bedard.

Now that all Apple port versions are 26, we can synthesize all of the different
CURRENT_VERSION definitions by port into one definition in DarwinPort. If we
have different versioning for whatever reason (maybe differences in minor
version?), we can always override that in the subclasses.

This change also changes associated unit tests to use the current version
rather than manually defining it every time, requiring fewer changes each year.

* Tools/Scripts/webkitpy/common/version_name_map.py:
    (VersionNameMap)
        -> (__init__): Pull CURRENT_VERSION from port objects for iOS, tvOS, watchOS, and visionOS.
                        (Not for macOS, as this requires a manual change to define new version names yearly anyway.)
* Tools/Scripts/webkitpy/port/darwin.py:
    (DarwinPort): Set default for CURRENT_VERSION to `Version(26)`.
* Tools/Scripts/webkitpy/port/ios.py:
    (IOSPort): Remove CURRENT_VERSION definition that overrides DarwinPort.
* Tools/Scripts/webkitpy/port/mac.py:
    (MacPort): Ditto.
* Tools/Scripts/webkitpy/port/visionos.py:
    (VisionOSPort): Ditto.
* Tools/Scripts/webkitpy/port/watch.py:
    (WatchPort): Ditto.
* Tools/Scripts/webkitpy/port/ios_device_unittest.py: Change unit tests to use CURRENT_VERSION from associated ports.
* Tools/Scripts/webkitpy/port/ios_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/ios_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/visionos_testcase.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_simulator_unittest.py: Ditto.
* Tools/Scripts/webkitpy/port/watch_testcase.py: Ditto.

Canonical link: https://commits.webkit.org/296346@main
